lib/access-context.js

```diff
@@ -143,19 +144,9 @@ AccessContext.prototype.getUserId = function() {
     // or permit to resolve a valid user model
     var userModel = this.registry.findModel(p.type);
     if (!userModel) continue;
-    if (userModel.modelName === 'User') {
+    if (userModel.prototype instanceof BaseUser) {
       return p.id;
     }
```

Comment:
`registry.getModel()` calls `registry.findModel()` and returns exactly the one model registered with the given name - see [lib/registry.js#L299](https://github.com/strongloop/loopback/blob/798ebfba814d189c65a4fcf6c5026a932e873f04/lib/registry.js#L299) in current master. Here is the code in juggler adding an entry to `modelBuilder.models`: [lib/model-builder.js#L203](https://github.com/strongloop/loopback-datasource-juggler/blob/c509a9f7fe48877a2fc6d39ad98bd9007e42a560/lib/model-builder.js#L203).

Perhaps you are confusing `registry.getModels()` with `app.models` or `registry.getModelByType`?